### DISCUSSION TITLE
getting InvalidArgumentException Syntax error

### DISCUSSION THREAD

**Discussion by amoktar** [laravel/framework#52327] [Q&A]
### Laravel Version

11.17.0

### PHP Version

8.2.12

### Database Driver & Version

MySQL

### Description

I've traced the error till I found it from the 11.17.0 and higher

I'm not sure it's a bug or code safety restriction

Thanks

### Steps To Reproduce

it occurs when returns the model or returns an anonymous class

```php
class PostController extends Controller
{
    public function show(Post $post)
    {
         // -> trigger 1: I used to return the model as a json like so
         return $post;
         
        // -> trigger 2: Also when return an anonymous class
        // but here getting different error : TypeError
       /// Symfony\Component\HttpFoundation\Response::setContent(): Argument #1 ($content) must be of type ?string, class@anonymous given, called in C:\Users\AMoktar\Documents\code\infoems-webapp\vendor\laravel\framework\src\Illuminate\Http\Response.php on line 73


        return new class{};
    }
}
```

**Comment by henzeb** [comment]
setContent expects a string, so either implement Stringable interface on the anonymous class or something like jsonSerializable. Not sure why that error shows up there though.

What is the exact error when returning $post?

**Reply by amoktar** [reply]
Thank you for joining me.

I get 
`InvalidArgumentException
Syntax error`

not clear. I had to dig to figure it. the error stack was pointing to VerifyCsrfToken middleware, 
I changed the return $post => return $post->getAttributes() to get over it
